src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/integTests/framework/Host.java

```diff
@@ -0,0 +1,454 @@
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.scheduler.RCAScheduler;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.scheduler.RcaSchedulerState;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.reader.ClusterDetailsEventProcessor;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.threads.ThreadProvider;
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.lang.reflect.InvocationTargetException;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.ProtocolException;
+import java.net.URL;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import org.apache.commons.io.FileUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.jooq.Record;
+import org.jooq.Result;
+import org.junit.Assert;
+
+/**
+ * This class simulates a cluster node that executes an RCA graph. Each node has its own
+ * - GRPC server,
+ * - web server
+ * - RCAController and everything that it starts.
+ */
+public class Host {
+  private static final Logger LOG = LogManager.getLogger(Host.class);
+  private final boolean useHttps;
+  /**
+   * Each host has its own AppContext instance.
+   */
+  private final AppContext appContext;
+  private final HostTag myTag;
+  /**
+   * This uniquely identifies a host.
+   */
+  private final int hostId;
+  /**
+   * For Integration tests, where all the virtual nodes are part of the same JVM, Ip string does not matter. But for
+   * the sake of having this value filled, the string is 127.0.0.(hostId).
+   */
+  private final String hostIp;
+  private final AllMetrics.NodeRole role;
+  private final int webServerPort;
+  private final int grpcServerPort;
+  private final File hostDir;
+  private final boolean rcaEnabled;
+  private GRPCConnectionManager connectionManager;
+  private ClientServers clientServers;
+  private ScheduledExecutorService netOperationsExecutor;
+  private RcaControllerIt rcaController;
+  private Thread grpcThread;
+  private Thread webServerThread;
+  private Thread rcaControllerThread;
+  private ThreadProvider threadProvider;
+  private Path rcaEnabledFile;
+
+  public Host(int hostId,
+              boolean useHttps,
+              AllMetrics.NodeRole role,
+              int httpServerPort,
+              int grpcServerPort,
+              File clusterDir,
```

Comment:
Can we simply pass the Cluster object into Host object ?

Reply:
This will disrupt the layered structure. The way it is that cluster has the data for all the hosts but hosts don't